src/optimizer/query_rewrite.c

```diff
@@ -5457,26 +5501,31 @@ qo_rewrite_outerjoin (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
 	    {
 	      if (spec->info.spec.join_type == PT_JOIN_LEFT_OUTER)
 		{
-		  info.id = spec->info.spec.id;
+		  info.id = info_spec.id = spec->info.spec.id;
 		}
 	      else if (prev_spec != NULL)
 		{
-		  info.id = prev_spec->info.spec.id;
+		  info.id = info_spec.id = prev_spec->info.spec.id;
```

Comment:
for else clause, not needed to set info.id?

Reply:
I fixed it